lib/BC/SleighLifter.cpp

```diff
@@ -0,0 +1,1292 @@
+      }
+      default: break;
+    }
+
+    return LiftStatus::kLiftedUnsupportedInstruction;
+  }
+
+
+  bool dump(EmittedPcode current_target,
+            std::optional<EmittedPcode> look_ahead_op) override {
+    LOG(INFO) << "inner handle" << std::endl;
+    llvm::IRBuilder bldr(this->target_block);
+
+    // The MULTIEQUAL op has variadic operands
+    if (current_target.opc == OpCode::CPUI_MULTIEQUAL ||
+        current_target.opc == OpCode::CPUI_CPOOLREF) {
+      this->UpdateStatus(
+          this->LiftVariadicOp(bldr, current_target.opc, current_target.outvar,
+                               current_target.vars, current_target.isize),
+          current_target.opc);
+      return false;
+    }
+
+    if (current_target.opc == OpCode::CPUI_CALLOTHER) {
+      if (current_target.isize == 3 &&
+          current_target.vars[0].offset < this->user_op_names.size() &&
+          this->user_op_names[current_target.vars[0].offset] == "claim_eq") {
+        LOG(INFO) << "Applying eq claim";
+        this->replacement_cont.ApplyEqualityClaim(
+            bldr, *this, current_target.vars[1], current_target.vars[2]);
+        return false;
+      }
+
+
+      this->UpdateStatus(LiftStatus::kLiftedUnsupportedInstruction,
+                         current_target.opc);
+      return false;
+    }
+
+    switch (current_target.isize) {
+      case 1: {
+        this->UpdateStatus(
+            this->LiftUnOp(bldr, current_target.opc, current_target.outvar,
+                           current_target.vars[0]),
+            current_target.opc);
+        break;
+      }
+      case 2: {
+        auto res = this->LiftBinOp(
+            bldr, current_target.opc, current_target.outvar,
+            current_target.vars[0], current_target.vars[1], look_ahead_op);
+        this->UpdateStatus(res.first, current_target.opc);
+        return res.second;
+      }
+      case 3: {
+        this->UpdateStatus(
+            this->LiftTernOp(bldr, current_target.opc, current_target.outvar,
+                             current_target.vars[0], current_target.vars[1],
+                             current_target.vars[2]),
+            current_target.opc);
+        break;
+      }
+      default:
+        //this->replacement_cont.ApplyNonEqualityClaim();
+        this->UpdateStatus(LiftStatus::kLiftedUnsupportedInstruction,
+                           current_target.opc);
+        return false;
+    }
+    return false;
+    //this->replacement_cont.ApplyNonEqualityClaim();
+  }
+
+  LiftStatus GetStatus() {
+    return this->status;
+  }
+};  // namespace remill
+
+std::unordered_set<OpCode>
+    SleighLifter::PcodeToLLVMEmitIntoBlock::INTEGER_COMP_OPS = {
+        CPUI_INT_EQUAL,   CPUI_INT_NOTEQUAL,  CPUI_INT_LESS,
+        CPUI_INT_SLESS,   CPUI_INT_LESSEQUAL, CPUI_INT_SLESSEQUAL,
+        CPUI_INT_SBORROW, CPUI_INT_SCARRY,    CPUI_INT_CARRY};
+
+std::map<OpCode, SleighLifter::PcodeToLLVMEmitIntoBlock::BinaryOperator>
```

Comment:
What about
```
struct MyBuilder
{
   using builder_t = llvm::IRBuilder;
   builder_t &irb;

   auto translate(auto opcode, const std::vector< llvm::Value * > &ops)
   {
      switch(opcode)
      {
        case CPUI_INT_AND: return simple_op< 2 >(&builder_t::CreateAnd, ops);
      }
   }
};

auto translate(opcode, bldr,  ops) { return MyBuilder(bldr).translate( ... ); } // or some other wrapper fn.
```
I feel this has less bolierplate and since it is a type, it can be easily swapped if someone desires so.

Reply:
This could be a new issue.